ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -1112,15 +865,15 @@ public void destroy() {
     boolean isTxnOpen = driverContext != null
         && driverContext.getTxnManager() != null
         && driverContext.getTxnManager().isTxnOpen();
-    if (!hiveLocks.isEmpty() || isTxnOpen) {
+    if (!driverTxnHandler.hasHiveLock() || isTxnOpen) {
```

Comment:
See comment above: DriverTxnHanlder.destroy?

Reply:
I agree, I'll modify this.

Reply:
I agree, I'll modify this.